app/src/common/shared/com/igalia/wolvic/telemetry/TelemetryService.java

```diff
@@ -175,7 +175,7 @@ public static void closeWindowEvent(int windowId) {
     }
 
     private static String getDefaultSearchEngineIdentifierForTelemetry() {
-        return SearchEngineWrapper.get(context).getCurrentSearchEngine().getIdentifier();
+        return SearchEngineWrapper.get(context).resolveCurrentSearchEngine().getId();
```

Comment:
Can we remove `getCurrentSearchEngine()`. If it's unsafe to use, let's better do not let people use it.